homeassistant/components/neurio_energy/sensor.py

```diff
@@ -137,11 +143,18 @@ def __init__(self, data, name, sensor_type, update_call):
         self.update_sensor = update_call
         self._state = None
 
+        self._attr_state_class = STATE_CLASS_MEASUREMENT
         if sensor_type == ACTIVE_TYPE:
             self._unit_of_measurement = POWER_WATT
-            self._attr_state_class = STATE_CLASS_MEASUREMENT
+            self._attr_device_class = DEVICE_CLASS_POWER
         elif sensor_type == DAILY_TYPE:
             self._unit_of_measurement = ENERGY_KILO_WATT_HOUR
+            self._attr_device_class = DEVICE_CLASS_ENERGY
+
+    @property
+    def last_reset(self) -> datetime:
+        """Return the time when the sensor was initialized."""
+        return dt_util.start_of_local_day()
```

Comment:
Only accumulating measurements like energy consumption should set last_reset.

https://developers.home-assistant.io/docs/core/entity/sensor#properties

Reply:
The `last_reset` attribute should return a UTC time (datetime) not a local time.

Reply:
> Only accumulating measurements like energy consumption should set last_reset.
> 
> https://developers.home-assistant.io/docs/core/entity/sensor#properties

Code is cut and paste from Wemo's sensor (`dt` was renamed to `dt_util` to match existing Neurio import).  

https://github.com/home-assistant/core/blob/4ef9269790170d94b4b8f15f591e9f9f08abfb24/homeassistant/components/wemo/sensor.py#L120-L123

> The `last_reset` attribute should return a UTC time (datetime) not a local time.

Being a daily total, it resets every day at midnight `00:00` local time.  Wouldn't converting to UTC would require calculating the TimeZone offset and adding it back to UTC time giving the same value as start_of_day?

Reply:
Looks like `start_of_local_day` already returns UTC time so that's good. I was a bit confused by the name and docstring.
```
Python 3.8.10 (default, Jun  2 2021, 10:49:15) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.24.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from homeassistant.util import dt as dt_util

In [2]: dt_util.start_of_local_day()
Out[2]: datetime.datetime(2021, 8, 11, 0, 0, tzinfo=datetime.timezone.utc)
```
We still need to distinguish between the power sensor and the energy sensor here. Only the energy sensor should set `last_reset`.

Reply:
> We still need to distinguish between the power sensor and the energy sensor here. Only the energy sensor should set `last_reset`.

Ok, I'll sort that out.

Reply:
Yeah, that's ok for now. A future improvement would be to use our EntityDescription class to describe the different sensor properties.

https://github.com/home-assistant/core/blob/98a4e6a7e8bf29fc8e17be57423ad590300dc177/homeassistant/components/sensor/__init__.py#L103

Reply:
Saw that one in the TPLink sensor; wanted to keep the changes as minimal as possible for speed of acceptance and since I don't really do much Python. 😄